mattermost/mattermost-plugin-autotranslate
# Fix and updated the plugin

#### Summary
- Updated minimum version to "5.22.0".
- Updated supported languages from AWS Translate.
- Translation is now supported, not only in reference to English but in any combination between source and target languages -- a _language pair_ supported by AWS Translate.
- Updated readme and code owner.
- Updated code setup - circleci setup, make file, eslint, etc.
- Updated the way it store objects. Previously it's modifying the `post` which is no longer permitted due to its immutability. Now, the store is saved as:
```
{
  ...
  plugins-autotranslate: {
    translatedPosts: {
      [post-id]: {/translation object/}
    },
    translations: {
      [translation-id]: {/translation object based on source/target language/}
    },
    userInfo: {/no change/}
  }
}
```

Test steps:
1. On initial run, `autotranslate` plugin is disabled to a user. Post a message and check that “Translate” is not available in post’s dropdown menu
2. See ephemeral help by posting `/autotranslate` or `/autotranslate help`. Check text, words or links for any discrepancy.
3. Enable the plugin for current user by `/autotranslate on` and see that it’s enabled with source: Auto and target: English.
4. Post any message in a different language [supported by AWS Translate](https://docs.aws.amazon.com/translate/latest/dg/what-is.html)
    - Check that “Translate” is available in that post. (Note: “Translate” is available to regular post only. Other post like system message should not have that option.)
    - Click “Translate” and see that the post is translated.
    - Click “close” to hide the translated message
5. Try to change different settings (info, source, target, of/off) and see if it works or translates as expected.
